### ISSUE TITLE
First login error

### ISSUE THREAD

**Issue by ifurther** [nix-community/NixOS-WSL#305]
## Bug description
Fresh install the version 23.5.5.2 and use `wsl -d NixOS`. The log should show it.

## To Reproduce
Please see logs

**Expected behavior**

## Logs
```
PS G:\nixos-wsl\23.05> wsl --import NixOS .\NixOS\ nixos-wsl.tar.gz --version 2
Import in progress, this may take a few minutes.
操作順利完成。
PS G:\nixos-wsl\23.05> wsl -d NixOS
<3>WSL (74) ERROR: CreateProcessEntryCommon:577: execvpe /run/current-system/sw/bin/bash failed 2
<3>WSL (74) ERROR: CreateProcessEntryCommon:586: Create process not expected to return
```

**Comment by nzbr** [comment]
What does `wsl.exe --version` show for you?

**Comment by ifurther** [comment]
`WSL version: 1.2.5.0
Kernel version: 5.15.90.1
WSLg version: 1.0.51
MSRDC version: 1.2.3770
Direct3D version: 1.608.2-61064218
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.19045.3448`

nzbr ***@***.***> 於 2023年10月7日 週六 上午4:58寫道：

**Comment by nzbr** [comment]
The latest WSL version is `2.0.4.0`. Try installing the [update through the Microsoft Store](https://www.microsoft.com/store/productid/9P9TQF7MRM4R?ocid=pdpshare) and see if that fixes it

**Comment by ifurther** [comment]
I installed again and the version is same
![store](https://github.com/nix-community/NixOS-WSL/assets/55025025/6a1abcaa-752a-408b-8c80-c3df0464f29d)

**Comment by SuperSandro2000** [comment]
I can't see a version number in the screenshots and I don't think any of us knows that asian language.

**Comment by ifurther** [comment]
Sorry the store can't change the language. My WSL version is 1.2.5.0, and the store doesn't have any upgrade.

**Comment by KSaiAkshit** [comment]
> Sorry the store can't change the language. My WSL version is 1.2.5.0, and the store doesn't have any upgrade.

There is an official guide by Microsoft on how to upgrade from WSL 1  to WSL 2.
https://learn.microsoft.com/en-us/windows/wsl/install#:~:text=Upgrade%20version%20from%20WSL%201%20to%20WSL%202

**Comment by nzbr** [comment]
@KSaiAkshit The guide is about upgrading from WSL 1 (emulated kernel, basically reverse-WINE) to WSL 2 (actual Hyper-V VM). That has, however, nothing to do with the version number from `wsl --version`. 
I really don't know who at Microsoft thought it was a good idea to start numbering WSL 2 releases at `0...` and later `1...`, but I _absolutely_ see how this is confusing

**Comment by KSaiAkshit** [comment]
Ahh, my bad my bad. Yeah just realized, it is quite confusing,

**Comment by kniteli** [comment]
The 2.0.4 release (now 2.0.6) is a pre-release. The advice on the README and here about wsl from the store seems to be outdated. 1.2.5 is the latest official release, and the only one you can get from the store. To upgrade to the latest in pre-release, run this on cli (pwsh or cmd):

`wsl --update --pre-release`

**Comment by jackyliu16** [comment]
I guess I had a similar problem. This is a (recording screen](https://drive.google.com/file/d/17xiO3NyRQPbo_nWQwU36zgGt85KwsBx9/view?usp=drive_link). 

 Basically, I can't open the system via wsl -d after successfully installing NixOS-WSL, and I don't see any error messages 

1. I have use store to update the version of wsl, and also try for wsl update, but seems --pre-release not work for me
2. I have try multiable verison of tarball, two of the latest release and a nixos tarball export from the system before reinstall (same computer)
3. My default wsl version is 2

**Comment by nzbr** [comment]
@jackyliu16 
What does `wsl --version` report for you? 
Have you tried the whole process in a non-admin console? (Not sure if that changes anything, but it might be worth a shot)
If that doesn't help, please use the working debian install to get the output of `sudo dmesg` after you try starting NixOS

**Comment by aikooo7** [comment]
This is because of native systemd.

Change in ``/etc/nixos/configuration.nix``:

From:

``wsl.nativeSystemd = true;``

To:

``wsl.nativeSystemd = false;``

Please let me know if this fixes the issue

**Comment by aikooo7** [comment]
I was in mind I did already report this bug and told the workaround here, but searching through the issues I didn't find it

**Comment by aikooo7** [comment]
> This is very likely because of native systemd (Was my issue).
> 
> Change in `/etc/nixos/configuration.nix`:
> 
> From:
> 
> `wsl.nativeSystemd = true;`
> 
> To:
> 
> `wsl.nativeSystemd = false;`
> 
> Please let me know if this fixes the issue

@KSaiAkshit @jackyliu16

**Comment by nzbr** [comment]
@KSaiAkshit @jackyliu16 @aikooo7 
Please keep in mind, that syschdemd is no longer supported and only left as an option for compatibiliy purposes. It will probably be removed entirely at some point in the future

**Comment by aikooo7** [comment]
> @KSaiAkshit @jackyliu16 @aikooo7 
> Please keep in mind, that syschdemd is no longer supported and only left as an option for compatibiliy purposes. It will probably be removed entirely at some point in the future

Thanks for the heads up, still is a good workaround until we can find why this happens.

**Comment by SuperSandro2000** [comment]
Closing since WSL 1 support is only left for backwards compatibility and if problems can be fixed by upgrading to 2.0, then that is the solution.
